### PR TITLE
Update subscribed accounts when connected site authorizations are modified

### PR DESCRIPTION
**Pull request by F-OBrien**
Addresses #1411

### CODE REVIEW

**Comment by F-OBrien** [comment]
I you want a UI for testing that uses subscribed accounts you can use https://portal.polymesh.live/ (Polymesh Portal - Testnet)

**Comment by TarikGul** [comment]
Thanks for the PR on the queue to review tomorrow!

**Comment by F-OBrien** [comment]
Side note (this might be a bug?) I did not see any restriction in the signing flow for authorized accounts for a specific URL. Authorized accounts are only checked/filtered out when querying or subscribing to accounts. As a result as long as a key is in the keyring any authorized site can trigger the extension to prompt for signing with it. It just might not have an easy way to know it's in the keyring. I noticed this when I removed an authorization for a key with the current extension release 0.49.1 but was still able to sign with it.

I didn't include it in this PR as I'd be concerned any changes to the signing flow could be breaking for downstream consumers of these packages.

**Review by bee344 — APPROVED** [review]
Looks good on my side, thanks. A note is that the renaming will break stuff downstream, so we need to mention it in the next release's notes.

File: packages/extension-base/src/background/handlers/State.ts [hunk 1]
```diff
@@ -271,7 +284,7 @@ export default class State {
     };
   };
 
-  public udateCurrentTabsUrl (urls: string[]) {
+  public updateCurrentTabsUrl (urls: string[]) {
```
Comment (TarikGul) [inline packages/extension-base/src/background/handlers/State.ts:287]: Just as a side note: When we release this - this should be mentioned as a BREAKING CHANGE - it's quite minor but nonetheless will be nice to have in the CHANGELOG and release message. I do like this though.

**Comment by TarikGul** [comment]
> Side note (this might be a bug?) I did not see any restriction in the signing flow for authorized accounts for a specific URL. Authorized accounts are only checked/filtered out when querying or subscribing to accounts. As a result as long as a key is in the keyring any authorized site can trigger the extension to prompt for signing with it. It just might not have an easy way to know it's in the keyring. I noticed this when I removed an authorization for a key with the current extension release 0.49.1 but was still able to sign with it.

@bee344 @ryanleecode @IkerAlus Can you guys help look into this - I think it would important to get some second eyes and clarity on this. :)

File: packages/extension-base/src/background/handlers/State.ts (hunk 1 above)
Comment (ryanleecode) [reply packages/extension-base/src/background/handlers/State.ts:287]: should preserve the original method and just mark it as deprecated and add the new one with the fixed name

File: packages/extension-base/src/background/handlers/Extension.ts [hunk 2]
```diff
@@ -529,7 +529,7 @@ export default class Extension {
   }
 
   private updateCurrentTabs ({ urls }: RequestActiveTabsUrlUpdate) {
-    this.#state.udateCurrentTabsUrl(urls);
+    this.#state.updateCurrentTabsUrl(urls);
```
Comment (ryanleecode) [inline packages/extension-base/src/background/handlers/Extension.ts:532]: should be the inverse. this function becomes the actual implementation. the misspelled function just calls this one

File: packages/extension-base/src/background/handlers/State.ts (hunk 1 above)
Comment (F-OBrien) [reply packages/extension-base/src/background/handlers/State.ts:294]: I've reintroduced `udateCurrentTabsUrl` and marked as deprecated.

**Comment by Tbaut** [comment]
> As a result as long as a key is in the keyring any authorized site can trigger the extension to prompt for signing with it. 

The idea about the authorization is to make sure 1) a user actually agrees to share their accounts and 2) they can choose which accounts they share (released recently). What you describe certainly sounds like a check that has been forgotten, as I don't see why we should accept signing request for non-shared account. Regarding the gravity, I wonder how it could be exploited though, as in, a website should trigger many signing requests, with many different accounts, to try to guess what account is present in the keyring.

TL;DR: it should be fixed, but it's not a huge issue unless I missed something

**Comment by TarikGul** [comment]
> > As a result as long as a key is in the keyring any authorized site can trigger the extension to prompt for signing with it.
> 
> The idea about the authorization is to make sure 1) a user actually agrees to share their accounts and 2) they can choose which accounts they share (released recently). What you describe certainly sounds like a check that has been forgotten, as I don't see why we should accept signing request for non-shared account. Regarding the gravity, I wonder how it could be exploited though, as in, a website should trigger many signing requests, with many different accounts, to try to guess what account is present in the keyring.
> 
> TL;DR: it should be fixed, but it's not a huge issue unless I missed something

I'll follow up on this by filing an issue today -  and then we can tackle this eventually - and make it a bit more tolerant.
